py/stackctrl.h

```diff
@@ -26,15 +26,20 @@
 #ifndef MICROPY_INCLUDED_PY_STACKCTRL_H
 #define MICROPY_INCLUDED_PY_STACKCTRL_H
 
-#include "py/mpconfig.h"
+#include "py/mpstate.h"
 
 void mp_stack_ctrl_init(void);
 void mp_stack_set_top(void *top);
 mp_uint_t mp_stack_usage(void);
 
 #if MICROPY_STACK_CHECK
 
-void mp_stack_set_limit(mp_uint_t limit);
+inline static void mp_stack_set_limit(mp_uint_t limit) {
+    assert(limit > MICROPY_STACK_CHECK_MARGIN); // Should be enforced by port
```

Comment:
Suggest using `MP_STATIC_ASSERT` (if possible here...).

Reply:
Not possible here, limit is not guaranteed to be compile-time constant.

However the compiler will be smart, if assertions are enabled and limit *is* constant then the assert will be elided if it passes, and if it fails then dead code elimination will conveniently remove everything that comes after. :laughing:

Reply:
Aah, right.